frontend/src/app/applications/actions/cancel-application.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ApplicationService } from '../../_services/application.service';
+import { AlertService } from '../../_services/alert.service';
+import { AuthenticationService } from '../../_services/authentication.service';
+
+@Component({
+  selector: 'app-cancel-application',
+  template: `<button *ngIf="
+      application.status !== 'Accepted'
+      && application.status !== 'Cancelled'
+      && authentication.user.role === 'user'"
+      class="usa-button cancel-button" (click)="cancelApplication()">{{ text }}</button>`
+})
+export class CancelApplicationComponent {
+  @Input() application: any;
+  @Input() text: string;
+  @Output() applicationCancelled: EventEmitter<any> = new EventEmitter<any>();
+
+  constructor(
+    private applicationService: ApplicationService,
+    private alertService: AlertService,
+    public authentication: AuthenticationService
+  ) {}
+
+  cancelApplication() {
+    const confirm = window.confirm('Are you sure you want to cancel this application?');
+    if (confirm) {
+      this.updateApplication();
+    }
+  }
+
+  convertCamelToHyphenCase(type: string) {
```

Comment:
It seems like this could be moved to a util controller so that it would be more likely to be reused.

Reply:
I created a util service and moved this out.